zrml/global-disputes/src/lib.rs

```diff
@@ -647,8 +785,88 @@ mod pallet {
             }
         }
 
-        fn is_started(market_id: &MarketIdOf<T>) -> bool {
-            <Winners<T>>::get(market_id).is_some()
+        fn does_exist(market_id: &MarketIdOf<T>) -> bool {
+            <GlobalDisputesInfo<T>>::get(market_id).is_some()
+        }
+
+        fn is_active(market_id: &MarketIdOf<T>) -> bool {
+            if let Some(gd_info) = <GlobalDisputesInfo<T>>::get(market_id) {
+                if let GdStatus::Active { add_outcome_end: _, vote_end: _ } = gd_info.status {
+                    return true;
+                }
+            }
+            false
+        }
+
+        fn start_global_dispute(
+            market_id: &MarketIdOf<T>,
+            initial_items: &[InitialItemOf<T>],
+        ) -> Result<u32, DispatchError> {
+            let market = T::MarketCommons::market(market_id)?;
+
+            ensure!(
+                <GlobalDisputesInfo<T>>::get(market_id).is_none(),
+                Error::<T>::GlobalDisputeAlreadyExists
+            );
+
+            ensure!(initial_items.len() >= 2, Error::<T>::AtLeastTwoOutcomesRequired);
+
+            for InitialItem { outcome, owner, amount } in initial_items {
+                ensure!(market.matches_outcome_report(outcome), Error::<T>::OutcomeMismatch);
+
+                match <Outcomes<T>>::get(market_id, outcome) {
+                    Some(mut outcome_info) => {
+                        let outcome_sum = outcome_info.outcome_sum.saturating_add(*amount);
+                        outcome_info.outcome_sum = outcome_sum;
+                        let mut owners = outcome_info
+                            .possession
+                            .get_shared_owners()
+                            .ok_or(Error::<T>::SharedPossessionRequired)?;
+                        owners.try_push(owner.clone()).map_err(|_| Error::<T>::MaxOwnersReached)?;
+                        outcome_info.possession = Possession::Shared { owners };
+                        Self::update_winner(market_id, outcome, outcome_info.clone());
+                        <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                    }
```

Comment:
This branch doesn't appear to be well tested. As far as I can tell, it only triggers in one test, `start_global_dispute_fails_if_max_owner_reached`, which is kind of an extreme scenario. I'm assuming that the scenario in which this triggers is if there are multiple appeals in court and thus multiple owners? Then the `MaxOwnersReached` depends on other config values in the runtime, which should probably be documented somewhere, and the case must be tested with more care.

Reply:
Let me think the new court implementation through in order to modify this specific API properly...

Reply:
What's the status on this? Maybe we should wait with this PR until the others are merged so this can get resolved?

Reply:
I configured the treasury to be the outcome owner of all appealed outcomes. 

https://github.com/zeitgeistpm/zeitgeist/blob/995eae36fc00abba361c2a7335b512a70a307d1f/zrml/court/src/lib.rs#L2132

Added tests here: 9c814acc0cb08d5b3d18c4f1b20304ec52b71df4.